prow/cmd/deck/main.go

```diff
@@ -487,8 +488,25 @@ func prodOnlyMain(cfg config.Getter, pluginAgent *plugins.ConfigAgent, o options
 
 	mux.Handle("/prowjob", gziphandler.GzipHandler(handleProwJob(prowJobClient)))
 
+	// We use the GH client to resolve GH teams when determining who is permitted to rerun a job.
+	// Also we need it to get PullRequest in order to access presubmit config for showing the
+	// PR history page.
+	secretAgent := &secret.Agent{}
+	if err := secretAgent.Start([]string{o.github.TokenPath}); err != nil {
+		logrus.WithError(err).Fatal("Error starting secrets agent.")
+	}
+	githubClient, err := o.github.GitHubClient(secretAgent, o.dryRun)
```

Comment:
Seems like we should make this optional. Right now this will blow up if they haven't configured deck with github creds, right?

Reply:
Yes, that is correct. However the result will be that the pr-history endpoint wont work if you have `inrepoconfig` enabled and no credentials put here. That is IMHO a weird dependency thats not very obvious. I'd rather just leave this and require ppl to make credentials available.

Reply:
I would rather pull pr-history into its own deployment then. People need to be ale to run deck with minimal hassle and be able to see what jobs are running. They should not need to set up a github application.

Some of Google's prow deployments do not ever access github, so forcing these to configure a github application makes no sense.

Reply:
Okay, what about we make the flag itself optional but keep it the starter.yaml?